src/default-client/index.ts

```diff
@@ -0,0 +1,840 @@
+      }
+    }
+
+    const { fieldTypeMap } = context;
+
+    fieldTypeMap.set(fieldPath, {
+      hasArguments: !!argumentsObjectMap,
+      hasDirectives: !!directives,
+      isEntity,
+      resourceValue,
+      typeName,
+    });
+  }
+
+  private static _socketsSupported(): boolean {
+    if (!process.env.WEB_ENV) return true;
+    return !!self.WebSocket;
+  }
+
+  private _cache: CacheManager;
+
+  private _cachemapOptions: CachemapArgsGroup = {
+    dataEntities: {
+      indexedDBOptions: {
+        databaseName: "handl-dataEntities-store",
+        objectStoreName: "dataEntities",
+      },
+      name: "handl-dataEntities",
+      redisOptions: { db: 0 },
+      use: { client: "indexedDB", server: "redis" },
+    },
+    dataPaths: {
+      indexedDBOptions: {
+        databaseName: "handl-dataPaths-store",
+        objectStoreName: "dataPaths",
+      },
+      name: "handl-dataPaths",
+      redisOptions: { db: 1 },
+      use: { client: "indexedDB", server: "redis" },
+    },
+    responses: {
+      indexedDBOptions: {
+        databaseName: "handl-responses-store",
+        objectStoreName: "responses",
+      },
+      name: "handl-responses",
+      redisOptions: { db: 2 },
+      use: { client: "indexedDB", server: "redis" },
+    },
+  };
+
+  private _defaultCacheControls: DefaultCacheControls = {
+    mutation: "max-age=0, s-maxage=0, no-cache, no-store",
+    query: "public, max-age=60, s-maxage=60",
+    subscription: "max-age=0, s-maxage=0, no-cache, no-store",
+  };
+
+  private _fetcher: FetchManager;
+  private _resourceKey: string = "id";
+  private _schema: GraphQLSchema;
+  private _subscriptionsEnabled: boolean = false;
+  private _subscriptionService: SubscriptionService;
+
+  constructor(args: ClientArgs) {
+    if (!isPlainObject(args)) {
+      throw new TypeError("Constructor expected args to be a plain object.");
+    }
+
+    const {
+      batch,
+      cachemapOptions,
+      defaultCacheControls,
+      fetchTimeout,
+      headers,
+      introspection,
+      resourceKey,
+      subscriptions,
+      url,
+    } = args;
+
+    const errors: TypeError[] = [];
+
+    if (!isPlainObject(introspection)) {
+      errors.push(new TypeError("Constructor expected introspection to be a plain object."));
+    }
+
+    if (!isString(url)) {
+      errors.push(new TypeError("Constructor expected url to be a string."));
+    }
+
+    if (errors.length) throw errors;
+
+    if (cachemapOptions && isPlainObject(cachemapOptions)) {
+      this._cachemapOptions = merge(this._cachemapOptions, cachemapOptions);
+    }
+
+    if (defaultCacheControls && isPlainObject(defaultCacheControls)) {
+      this._defaultCacheControls = { ...this._defaultCacheControls, ...defaultCacheControls };
+    }
+
+    this._fetcher = new FetchManager({ batch, fetchTimeout, headers, url });
+    if (isString(resourceKey)) this._resourceKey = resourceKey;
+    const introspectionQuery = introspection as IntrospectionQuery;

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4325)